include/fiction/algorithms/simulation/sidb/random_layout_generator.hpp

```diff
@@ -0,0 +1,178 @@
+ * layout).
+ * @return A randomly generated layout of SiDBs.
+ */
+template <typename Lyt>
+Lyt generate_random_layout(const random_layout_params<Lyt>& params, const Lyt& lyt_skeleton)
+{
+    static_assert(is_cell_level_layout_v<Lyt>, "Lyt is not a cell-level layout");
+
+    const uint64_t number_of_sidbs_of_final_layout = lyt_skeleton.num_cells() + params.number_of_sidbs;
+
+    Lyt lyt{};
+
+    if (lyt_skeleton.num_cells() != 0)
+    {
+        lyt_skeleton.foreach_cell([&lyt, &lyt_skeleton](const auto& cell)
+                                  { lyt.assign_cell_type(cell, lyt_skeleton.get_cell_type(cell)); });
+    }
+
+    bool     successful_generation = false;
+    uint64_t attempt_counter       = 0;
+
+    // Stops if either all SiDBs are placed or the maximum number of attempts were performed.
+    while (lyt.num_cells() < number_of_sidbs_of_final_layout && attempt_counter < params.maximal_attempts)
+    {
+        const auto random_coord = random_coordinate(params.coordinate_pair.first, params.coordinate_pair.second);
+
+        bool constraint_violation_positive_sidbs = false;
+
+        if (params.prevent_positive_charges)
+        {
+            // Checks if the new coordinate is not closer than 2 cells (Euclidean distance) from an already
+            // placed SiDB.
+            lyt.foreach_cell(
+                [&lyt, &random_coord, &constraint_violation_positive_sidbs, &params](const auto& c1)
+                {
+                    if (euclidean_distance<Lyt>(lyt, c1, random_coord) < params.minimal_spacing)
+                    {
+                        constraint_violation_positive_sidbs = true;
+                    }
+                });
+        }
+        // If the constraint that no positive SiDBs occur is satisfied, the SiDB is added to the layout.
+        if (!constraint_violation_positive_sidbs)
+        {
+            lyt.assign_cell_type(random_coord, Lyt::cell_type::NORMAL);
+        }
+        attempt_counter += 1;
+    }
+
+    return lyt;
+}
+
+/**
+ * Generates multiple unique random layouts of SiDBs based on the provided parameters.
+ *
+ * @tparam Lyt The layout type.
+ * @param params The parameters for generating the random layouts.
+ * @param lyt_skeleton A layout to which random cells are added (useful if you need to add random cells to a given
+ * layout).
+ * @param number_of_unique_generated_layouts The desired number of unique layouts to be generated.
+ * @param maximal_attempts The maximum number of attempts allowed to generate a unique layout (default: \f$ 10^{6} \f$).
+ * @return A vector containing the unique randomly generated layouts.
+ */
+template <typename Lyt>
+std::vector<Lyt> generate_multiple_random_layout(const random_layout_params<Lyt>& params, const Lyt& lyt_skeleton,
+                                                 const uint64_t number_of_unique_generated_layouts,
+                                                 const uint64_t maximal_attemps = 10E6)
```

Comment:
Isn't the `maximal_attempts` parameter already present in the `params`?

Reply:
Good point! However, they are indeed different. Here, it describes the maximum number of attempts allowed to generate the given number of unique layouts. I have to update the docu above.